static/js/subs.js

```diff
@@ -81,6 +81,21 @@ function should_list_all_streams() {
     return !page_params.realm_is_zephyr_mirror_realm;
 }
 
+// this finds the stream that is actively open in the settings and focused in
+// the left side.
+exports.active_stream = function () {
+    var hash_components = window.location.hash.substr(1).split(/\//);
+
+    // if the string casted to a number is valid, and another component
+    // after exists then it's a stream name/id pair.
+    if (!isNaN(+hash_components[1]) && hash_components[2]) {
+        return {
+            id: +hash_components[1],
```

Comment:
What's the story with the `+`s here; is that casting it to a number?  

I worry a bit that this code path is not going to be super robust.  I think we should be having this interact with some library code that we put in `hashchange.js` that results in this only trying to parse the hash if it's actually a streams page narrow at all.

Reply:
Yep it's shorthand for casting to a number and works perfectly fine as a `parseFloat` analog:

```js
+"-Infinity" // -Infinity
+"-45.67"    // -45.67
+"13"        // 13
```

There may be value in having a central store that controls the streams and then updates the hash instead.

Reply:
OK.  We should just use parseFloat.  It's important for us to avoid making our contributors need to know every detail of how JS works to read our code (especially features like this that are hard to google).  We ban all sorts of similar things in our Python codebase to keep the codebase accessible to a wide audience of developers, and the same reasoning applies here.

Reply:
👍 you're right.